src/app/use-cases/request-signature.ts

```diff
@@ -39,10 +42,20 @@ export const makeRequestSignature =
           TE.chainW(TE.fromOption(() => productNotFoundError)),
           TE.chainEitherK(getDocumentsByMetadata)
         ),
+        expiresAt: payload.expiresAt
+          ? isBefore(payload.expiresAt, new Date())
+            ? TE.left(
+                new InvalidEntityError(
+                  "The expiration date must be in the future"
+                )
+              )
+            : TE.right(payload.expiresAt)
+          : TE.right(undefined),
```

Comment:
```suggestion
        expiresAt: pipe(
          TE.of(payload.expiresAt),
          TE.filterOrElse(
            (_) => _ !== undefined && isBefore(_, Date.now()),
            () =>
              new InvalidEntityError(
                "The expiration date must be in the future"
              )
          )
        ),
```

Reply:
you may consider to extract the predicate lambda into it's own method or to revert to its own type (FutureDate) to make the pipe terser ie. 

```
 expiresAt: pipe(
          TE.of(payload.expiresAt),
          TE.filterOrElse(
            FutureDate.is, // or isDateInTheFuture
            constant(
              new InvalidEntityError(
                "The expiration date must be in the future"
              ))
          )
        ),
```

Reply:
but `payload.expiresAt` could also be **undefined** (since it is optional). Could it be correct to write such a thing?
```
export function isFutureDate(date: UTCISODateFromString | undefined) {
  return date === undefined || isAfter(date, new Date());
}
```

Reply:
sorry you're right, just keep the check inline:

```
        expiresAt: pipe(
          TE.of(payload.expiresAt),
          TE.filterOrElse(
            (_) => _ === undefined || isAfter(_, Date.now()),
            constant(
              new InvalidEntityError(
                "The expiration date must be in the future"
              )
            )
          )
        ),

```